src/server/rest/v1/router/auth/AuthRouter.ts

```diff
@@ -38,8 +42,32 @@ export default class AuthRouter {
   }
 
   protected buildRoutePasswordReset(): void {
-    this.router.post(`/${ServerAction.REST_PASSWORD_RESET}`, async (req: Request, res: Response, next: NextFunction) => {
-      await RouterUtils.handleServerAction(AuthService.handleUserPasswordReset.bind(this), ServerAction.REST_PASSWORD_RESET, req, res, next);
+    this.router.post(`/password/${ServerAction.REST_RESET}`, async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Use full path: REST_PASSWORD_RESET and not REST_RESET
The code is not readable and the logs' actions won't be either.
Everything after 'v1/auth' should be a REST constant.

Reply:
Well, that's not what I asked Loan to do ;)
I wanted to distinguish data structure from action on them: 

Data structures: 
- mail
- password 
- tenants
- users
- ...

Actions: 
- read
- update
- create
- delete
- reset
- resend
- signin 
- ...

It's a REST endpoint URI construction pattern: ../{data)/{action} for requests, especially on GET. The action is not given when the action type is defined by the request types: DELETE, UPDATE, .... But when it does not exist or match, the action is added at the end of the URI path.
I think the entity enum is the closest match for URI data part. But at the beginning, I've told Loan to hardcode them and after more endpoints conversion to the new API, to create or reuse an enum for the data structure name.
Loan can create a log prefix string helper for REST endpoints usage:
```
logPrefix(data, action) {
  switch (action) {
	case 'GET'
		action = ServerAction.READ
		break
    ...	 
  return `${action} on ${data}:` 
} 
```

Reply:
@jerome-benoit 
I understand your point and it makes sense but the implementation is not correct then.

As you wrote:
- ServerAction should only contain the nouns: email, signin... and not actions
- Action should contain the action: check, reset...
- Handle the Logs Action + Noun: Check Email

I see these issues:
- Action is not lowercase but we can call toLowercase() on it to register the path
- Action filter in Logs is based on ServerAction only, now if we combine both ServerAction and Action, we will may more filters than necessary.

Let me know what you think

Reply:
I see the issue: there was some overlaps with Action and ServerAction. That's why I've started to use ServerAction for the new REST API URI path. But Action is really what I wanted in fact. 

So I think we should first: 

+ Untangle Action and ServerAction;
+ See if it makes sense to allow logging to use Action and combine them in logs filtering 

Or

+ Explicitly separate Action overlaps in ServerAction enum that will be used in REST endpoints and flag them with a prefix like REST_ACTION_... and a comment. It's the easiest solution.

Reply:
> I see the issue: there was some overlaps with Action and ServerAction. That's why I've started to use ServerAction for the new REST API URI path. But Action is really what I wanted in fact.
> 
> So, I think we should first:
> 
> * Untangle Action and ServerAction;
> * See if it makes sense to allow logging to use Action and combine them in logs filtering
> 
> Or
> 
> * Explicitly separate Action overlaps in ServerAction enum that will be used in REST endpoints and flag them with a prefix like REST_ACTION_... and a comment. It's the easiest solution.

So for the easiest solution will it be for 'reset password': ServerAction.REST_RESET_PASSWORD (='password/reset')?

Reply:
> > * Explicitly separate Action overlaps in ServerAction enum that will be used in REST endpoints and flag them with a prefix like REST_ACTION_... and a comment. It's the easiest solution.
> 
> So for the easiest solution will it be for 'reset password': ServerAction.REST_RESET_PASSWORD (='password/reset')?

No, because you do the reset action on the password data. Just like you do the resend action on mail data. And in ServerAction, we only put the REST related actions with a prefix. 
In that case, we have to add an entity if it miss and the URL is then .../{Entity.}/{ServerAction.}.
If we add the entity handling in the REST services callback, we're Ok for proper logging: we already have the HTTP methods and if the ServerAction is provided, we can make it have precedence over HTTP methods. 
But having specific actions on specific datas is really narrowed to the auth endpoint for now. 
And I think that taking a bit of time to do it properly will help because we do not know if in the future, we will need more than CRUD actions on other endpoints, especially for charging stations that are also special . 

Just like REST endpoints URI path hierarchy is very often mapped to the data hierarchy: 

/tenants -> CRUD via HTTP methods
/tenants/organisations -> CRUD via HTTP methods
/tenants/organisations/sites -> CRUD via HTTP methods
/tenants/organisations/sites/siteareas -> CRUD via HTTP methods

Here the REST endpoints URI path do not also match one entity but we already have the express router hierarchy that will match. 

Since there's no real urgence to implement to new REST API, we can probably do meetings to discuss it to ensure we do it right. The reference on REST ressource naming is https://restfulapi.net/resource-naming/. The specific action are here named controller.

Reply:
> /tenants -> CRUD via HTTP methods
> /tenants/organisations -> CRUD via HTTP methods
> /tenants/organisations/sites -> CRUD via HTTP methods
> /tenants/organisations/sites/siteareas -> CRUD via HTTP methods

There's no organizations entity, so it's more like this I think:

In Master Tenant:
/tenants
/tenants/1234

Within the tenant:
/companies
/companies/1234/sites
/companies/1234/sites/1234/siteareas
/companies/1234/sites/1234/siteareas/1234

I agree with you to build an endpoint with the Entity and Action: 'password' + 'reset'

But if we want to keep the ServerAction's current behavior which is referencing the endpoints (entities + actions) and being used in the logs' action filter, we can have in ServerAction:
REST_PASSWORD_RESET = `${Entity.EMAIL}/${Action.CHECK}`

And we use REST_PASSWORD_RESET in the routing.

In the end there are 2 aspects:
- External usage of REST API: '/password/reset' which we both agree
- Internal routing construction of this REST API which we have to agree :-)

Reply:
> 
> I agree with you to build an endpoint with the Entity and Action: 'password' + 'reset'
> 
> But if we want to keep the ServerAction's current behaviour which is referencing the endpoints (entities + actions) and being used in the logs' action filter, we can have in ServerAction:
> REST_PASSWORD_RESET = `${Entity.EMAIL}/${Action.CHECK}`
> 
> And we use REST_PASSWORD_RESET in the routing.
> 

That's also fine as long as everything is lower cased as ServerAction building. If we pass that ServerAction to the logging code, we mix the action and the entity, which is not really clean but not a big deal.